examples/runners/ash/src/main.rs

```diff
@@ -756,92 +720,119 @@ impl RenderCtx {
-        let frag_name = CString::new(fs_entry_point).unwrap();
-        let desc = PipelineDescriptor::new(Box::new([
-            vk::PipelineShaderStageCreateInfo {
-                module: *vert_module,
-                p_name: (*vert_name).as_ptr(),
-                stage: vk::ShaderStageFlags::VERTEX,
-                ..Default::default()
-            },
-            vk::PipelineShaderStageCreateInfo {
-                s_type: vk::StructureType::PIPELINE_SHADER_STAGE_CREATE_INFO,
-                module: *frag_module,
-                p_name: (*frag_name).as_ptr(),
-                stage: vk::ShaderStageFlags::FRAGMENT,
-                p_specialization_info: &specialization_info,
-                ..Default::default()
-            },
-        ]));
-        let desc_indirect_parts = desc.indirect_parts();
-        let pipeline_info = vk::GraphicsPipelineCreateInfo::default()
-            .stages(&desc.shader_stages)
-            .vertex_input_state(&desc.vertex_input)
-            .input_assembly_state(&desc.input_assembly)
-            .rasterization_state(&desc.rasterization)
-            .multisample_state(&desc.multisample)
-            .depth_stencil_state(&desc.depth_stencil)
-            .color_blend_state(&desc_indirect_parts.color_blend)
-            .dynamic_state(&desc_indirect_parts.dynamic_state_info)
-            .viewport_state(&viewport)
-            .layout(pipeline_layout)
-            .render_pass(self.render_pass);
-
-        let mut pipelines = unsafe {
-            self.base
+            let pipeline_layout = self
+                .base
                 .device
-                .create_graphics_pipelines(pipeline_cache, &[pipeline_info], None)
-                .expect("Unable to create graphics pipeline")
-        };
-        // A single `pipeline_info` results in a single pipeline.
-        assert_eq!(pipelines.len(), 1);
-        self.pipeline = pipelines.pop().map(|pipeline| Pipeline {
-            pipeline,
-            pipeline_layout,
-        });
+                .create_pipeline_layout(
+                    &vk::PipelineLayoutCreateInfo::default().push_constant_ranges(&[
+                        vk::PushConstantRange::default()
+                            .offset(0)
+                            .size(size_of::<ShaderConstants>() as u32)
+                            .stage_flags(vk::ShaderStageFlags::ALL),
+                    ]),
+                    None,
+                )
+                .unwrap();
+
+            let mut pipelines =
+				self.base
+					.device
+					.create_graphics_pipelines(vk::PipelineCache::null(), &[vk::GraphicsPipelineCreateInfo::default()
+						.stages(
+							&[
+								vk::PipelineShaderStageCreateInfo {
+									module: shader_module,
+									p_name: c"main_vs".as_ptr(),
+									stage: vk::ShaderStageFlags::VERTEX,
+									..Default::default()
+								},
+								vk::PipelineShaderStageCreateInfo {
+									module: shader_module,
```

Comment:
I see, you can use `shader_module` for multiple stages; the differing entry points distinguish them. :+1: